core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -431,7 +439,10 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
     // take into account executors that are pending to be added or removed.
     if (!replace) {
       doRequestTotalExecutors(
-        numExistingExecutors + numPendingExecutors - executorsPendingToRemove.size)
+        numExistingExecutors + numPendingExecutors - executorsPendingToRemove.size
+        + numReplacingExecutors)
+    } else {
+      numReplacingExecutors += knownExecutors.size
```

Comment:
do we need another variable? Can't we just do
```
numPendingExecutors += knownExecutors.size
```
This makes sense on a high level too; if we replace an executor we expect to get one back, so it should be pending in the mean time.